### PR TITLE
Puredata with plugins

### PR DESCRIPTION
**Pull request by magnetophon**

### CODE REVIEW

**Comment by jagajaga** [comment]
Rebase?

**Comment by magnetophon** [comment]
@jagajaga rebase done.

File: pkgs/applications/audio/pd-plugins/cyclone/default.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  src = fetchurl {
+    url = "http://downloads.sourceforge.net/project/pure-data/libraries/cyclone/${name}.tar.gz";
+    sha256 = "1yys9xrlz09xgnqk2gqdl8vw6xj6l9d7km2lkihidgjql0jx5b5i";
+  };
+
+  buildInputs = [ puredata ];
+
+  /*unpackPhase = ''*/
+    /*unzip $src*/
+  /*'';*/
```
Comment (pSub) [inline pkgs/applications/audio/pd-plugins/cyclone/default.nix]: What does uncommented unpackPhase? If it isn't needed, please remove it.

File: pkgs/applications/audio/pd-plugins/cyclone/default.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      done
+  '';
+
+  installPhase = ''
+    mkdir -p $out/cyclone
+    cp -r bin/* $out/cyclone
+  '';
+
+  meta = {
+    description = "A library of PureData classes, bringing some level of compatibility between Max/MSP and Pd environments.";
```
Comment (pSub) [inline pkgs/applications/audio/pd-plugins/cyclone/default.nix]: Descriptions should not end with a dot. See http://nixos.org/nixpkgs/manual/#chap-meta

File: pkgs/applications/audio/pd-plugins/helmholtz/default.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    sed -i "s@current: pd_darwin@current: pd_linux@g" Makefile
+    sed -i "s@-Wl@@g" Makefile
+  '';
+
+  installPhase = ''
+    cp -r helmholtz~/ $out/
+  '';
+
+  meta = {
+    description = "time domain pitch tracker for Pure Data";
```
Comment (pSub) [inline pkgs/applications/audio/pd-plugins/helmholtz/default.nix]: Descriptions should start with a capital letter.

File: pkgs/applications/audio/pd-plugins/maxlib/default.nix [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    sed -i "s@/usr@$out@g" Makefile
+  '';
+
+  postInstall = ''
+    mv $out/local/lib/pd-externals/maxlib/ $out
+    rm -rf $out/local/
+  '';
+
+  meta = {
+    description = "A library of non-tilde externals for puredata, by Miller Puckette.";
```
Comment (pSub) [inline pkgs/applications/audio/pd-plugins/maxlib/default.nix]: Again, no dots in `description`.

File: pkgs/applications/audio/pd-plugins/mrpeach/default.nix [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  '';
+
+  fixupPhase = ''
+    mv $out/lib/pd-externals/net $out
+    mv $out/lib/pd-externals/osc $out
+    rm -rf $out/lib
+  '';
+
+  meta = {
+    description = "A collection of Pd objectclasses for OSC-messages.";
```
Comment (pSub) [inline pkgs/applications/audio/pd-plugins/mrpeach/default.nix]: Same here.

File: pkgs/applications/audio/pd-plugins/puremapping/default.nix [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    unzip $src
+  '';
+
+  installPhase = ''
+    mkdir -p $out/puremapping
+    mv puremapping/ $out
+  '';
+
+  meta = {
+    description = "Set of externals to facilitate the use of sensors within Pure Data and to create complex relations between input and output of a dynamic system.";
```
Comment (pSub) [inline pkgs/applications/audio/pd-plugins/puremapping/default.nix]: And here.

File: pkgs/applications/audio/pd-plugins/zexy/default.nix [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    ./configure --enable-lpt=no --prefix=$out
+  '';
+
+  postInstall = ''
+    mv $out/lib/pd/extra/zexy $out
+    rm -rf $out/lib
+  '';
+
+  meta = {
+    description = "the swiss army knife for puredata";
```
Comment (pSub) [inline pkgs/applications/audio/pd-plugins/zexy/default.nix]: Capital letter.

**Comment by magnetophon** [comment]
@pSub done. Should I rebase again?

**Comment by lethalman** [comment]
Travis passed, thanks.

**Pull request merged**